tests/unit/script_plugins/lib.rs

```diff
@@ -32,6 +45,24 @@ pub mod unrooted_must_root {
     */
     pub fn struct_field() {}
 
+    /**
+    ```compile_fail
+    #![feature(plugin)]
+    #![plugin(script_plugins)]
+    #![feature(generic_param_attrs)]
+
+    #[must_root] struct Foo(i32);
+    struct Bar<#[must_root] T>(T);  // is not detected
```

Comment:
Even though I have written it explicitly, the `check_struct_def` hasn't detected it.

Reply:
this is presumably because `has_attr` doesn't work -- in particular, it will spuriously return false for `TyParam` until it is fixed (we talked about the fix on gitter)